hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java

```diff
@@ -292,7 +294,7 @@ public long getIdNumber(@NonNull Address address) {
         }
 
         if (account.deleted()) {
```

Comment:
Here we definitely want to just return `null` (i.e. to the calling code a deleted account appears to be just missing---as it would be if the `selfdestruct` had been executed in Ethereum).

Reply:
@tinker-michaelj I agree with the `HandleException` I throw it by default 😄

I think we should throw some error or have a way to differentiate between "deleted account" and "not found account/token"(see the PR "Notes for reviewer" section for details).

Maybe in ProxyWorldUpdated.getHederaAccount when we catch the `IllegalArgumentException` we shouldn't return null?

In the current implementation `factoryAndSelfDestructInConstructorContract` test is failing because it expects `CONTRACT_DELETED` but we return `INVALID_CONTRACT_ID`

Reply:
Ah got it; we should then add a top-level check in `HevmTransactionFactory.assertValidCall()` [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/HevmTransactionFactory.java#L239) that fails "explicitly" on the deleted contract.

But afaict this method needs to have Besu EVM semantics; that is, return `null` when a usable contract does not exist, but don't throw unchecked exceptions.

Reply:
Makse sense 👍 Fixed.